src/solidity/utils.ts

```diff
@@ -32,26 +32,35 @@ export function solidifyName(fileName: string): string {
 
 export async function gatherSources(
   what: string,
-  workingDir?: string,
+  workingDir: string = process.cwd(),
   resolver: ResolverEngine<ImportFile> = SolidityImportResolver(),
 ): Promise<ImportFile[]> {
   let result: ImportFile[] = [];
-  let queue = [];
+  let queue: { cwd: string; file: string }[] = [];
   let alreadyImported = new Set();
 
-  queue.push({ cwd: workingDir, file: what });
-  alreadyImported.add(solidifyName(what));
+  const absoluteWhat = path.resolve(workingDir, what);
+  queue.push({ cwd: workingDir, file: absoluteWhat });
+  alreadyImported.add(solidifyName(absoluteWhat));
```

Comment:
I vividly remember our discussion regarding this algorithm, and I suggested more comments, explaining why this trick works and doesn't loop for example (I know it's because of initial path being an absolute path, but I know it *now*, I might forget this fact in the future and so may you).